src/base-input.js

```diff
@@ -734,14 +750,31 @@ export default class BaseInput extends LitElement {
 
       if (!pattern.test(this.value)) {
         this.validity = 'badInput';
-        this.setCustomValidity = this.setCustomValidityBadInput;
+        this.setCustomValidity = this.setCustomValidityBadInput || '';
       }
     } else if (this.value.length > 0 && this.value.length < this.minLength) {
       this.validity = 'tooShort';
-      this.setCustomValidity = this.setCustomValidityTooShort;
+      this.setCustomValidity = this.setCustomValidityTooShort || '';
     } else if (this.value.length > this.maxLength) {
       this.validity = 'tooLong';
-      this.setCustomValidity = this.setCustomValidityTooLong;
+      this.setCustomValidity = this.setCustomValidityTooLong || '';
+    }
+  }
+
+  /**
+   * Sets configuration data used elsewhere based on the `type` attribute.
+   * @private
+   * @returns {void}
+   */
+  configureDataForType() {
+    if (this.type === 'month-day-year') {
+      this.dateStrLength = 10;
+    } else if (this.type === 'month-year') {
+      this.dateStrLength = 5;
+    } else if (this.type === 'month-fullYear') {
+      this.dateStrLength = 7;
+    } else if (this.type === 'year-month-day') {
```

Comment:
Add this to the first `if` statement: `if (this.type === ('month-day-year' || 'year-month-day')) {